### PR TITLE
Shelf use full width.

### PR DESCRIPTION
**Pull request by JuanAlmeida12**
#### What is the purpose of this pull request?
`Shelf` use full width when `itemsPerPage` are five or more and padding between items is set using tachyons. Also add gap to Shelf schema.

#### How should this be manually tested?

[Access this Workspace](https://galleryspace--storecomponents.myvtex.com)

#### Screenshots or example usage

galleryspace--storecomponents myvtex com_ 9
galleryspace--storecomponents myvtex com_ 10

#### Types of changes
- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/ShelfContent.js [hunk 1]
```diff
@@ -76,27 +78,35 @@ class ShelfContent extends Component {
   }
 
   get itemsToShow() {
-    const { itemsPerPage, width, gap } = this.props
-    const maxItems = Math.floor(width / (DEFAULT_SHELF_ITEM_WIDTH + gap))
+    const { itemsPerPage, width } = this.props
+    const maxItems = Math.floor(width / (DEFAULT_SHELF_ITEM_WIDTH))
```
Comment (GustavoMA) [inline react/ShelfContent.js]: Removing the gap from the math won't make the possibility to have more itens that can fit in the window?
Like, could only fit 3 but somehow got 4 from this math?

Comment (GustavoMA) [reply react/ShelfContent.js]: I know you changed the gap to be a tachion-like prop, but you can map it to the width and use in the math.

File: react/ProductList.js [hunk 2]
```diff
@@ -100,6 +102,13 @@ ProductList.getSchema = props => {
         default: ProductList.defaultProps.maxItems,
         isLayout: true,
       },
+      gap: {
+        title: 'editor.shelf.gap.title',
+        type: 'string',
+        enum: ['0x', '1x', '2x', '3x'],
```
Comment (theomoura) [inline react/ProductList.js]: You should use your enum values here and not repeat the information of it. Create a enum file or export it.

File: react/ProductList.js [hunk 3]
```diff
@@ -140,6 +149,7 @@ ProductList.defaultProps = {
   maxItems: DEFAULT_MAX_ITEMS,
   itemsPerPage: DEFAULT_ITEMS_PER_PAGE,
   scroll: ScrollTypes.BY_PAGE.value,
+  gap: '1x',
```
Comment (theomoura) [inline react/ProductList.js]: Use your enum here

File: react/ShelfContent.js [hunk 4]
```diff
@@ -76,27 +78,35 @@ class ShelfContent extends Component {
...
+    if (items >= ITEMS_TO_FULL_WIDTH || width <= slider) return width
+
+    return slider
   }
 
-  slideFallback = (item = {}, key) => {
-    const { summary, gap } = this.props
+  get paddingForItem() {
+    const { gap } = this.props
+    return GAP_TYPES[gap] || 'ph0'
```
Comment (theomoura) [inline react/ShelfContent.js]: The `ph0` padding shouldn' be an option on gap, like `gap: none`

File: react/ShelfContent.js (hunk 1 above)
Comment (JuanAlmeida12) [reply react/ShelfContent.js]: Gap now are paddings(before are margins),  so this we do not have this problem.

File: react/ShelfContent.js [hunk 5]
```diff
@@ -175,8 +193,8 @@ ShelfContent.propTypes = {
   summary: PropTypes.any,
   /** Is mobile */
   isMobile: PropTypes.bool,
-  /** Gap in Px between Shelf Items */
-  gap: PropTypes.number,
+  /** Gap between Shelf Items */
+  gap: PropTypes.oneOf(['0x', '1x', '2x', '3x']),
```
Comment (theomoura) [inline react/ShelfContent.js]: Could use enum here as well.

File: react/paddingEnum.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    name: 'editor.shelf.gapType.medium',
+    value:'ph5',
+  },
+  LARGE: {
+    name: 'editor.shelf.gapType.large',
+    value:'ph7',
+  },
+}
+
+export const getGapPaddingNames = () => {
```
Comment (GustavoMA) [inline react/paddingEnum.js]:
```suggestion
export const getGapPaddingNames = () => pluck('name', gapPaddingTypes)
```

File: react/paddingEnum.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+  const names = []
+  for (const key in gapPaddingTypes) names.push(gapPaddingTypes[key].name)
+  return names
+}
+
+export const getGapPaddingValues = () => {
```
Comment (GustavoMA) [inline react/paddingEnum.js]:
```suggestion
export const getGapPaddingValues = () => pluck('value', gapPaddingTypes)
```

File: react/paddingEnum.js (hunk 6 above)
Comment (GustavoMA) [reply react/paddingEnum.js]: and import ramda's pluck function.

File: react/paddingEnum.js (hunk 7 above)
Comment (GustavoMA) [reply react/paddingEnum.js]: and import ramda's pluck function.

**Pull request merged**